eras/shelley/impl/testlib/Test/Cardano/Ledger/Shelley/ImpTest.hs

```diff
@@ -470,56 +479,61 @@ fixupFees tx = do
       mkBasicTxOut @era
         (mkAddr (kpSpending, kpStaking))
         (inject remainingCoin)
-  impRootTxCoinL .= remainingCoin
   let
-    balancedTx =
-      setMinFeeTx pp tx
+    txNoWits =
+      tx
         & bodyTxL . inputsTxBodyL %~ Set.insert (TxIn impRootTxId $ TxIx 0)
         & bodyTxL . outputsTxBodyL %~ (remainingTxOut :<|)
-    txId = TxId . hashAnnotated $ balancedTx ^. bodyTxL
-  impRootTxIdL .= txId
-  wits <- mkTxWits (balancedTx ^. bodyTxL)
-  pure $ balancedTx & witsTxL .~ wits
+    nativeScriptKeyWitsCount = 0 -- TODO figure out wits for native scripts
+    outsWithoutRemaining = tx ^. bodyTxL . outputsTxBodyL
+    fee = calcMinFeeTx utxo pp txNoWits nativeScriptKeyWitsCount
+    remainderAvailable = remainingTxOut ^. coinTxOutL <-> fee
+    remainingTxOut' = remainingTxOut & coinTxOutL .~ remainderAvailable
```

Comment:
We could define this as 
`remainingTxOut & coinTxOutL %~ (<-> fee)`, but maybe you wanted to emphasize the relationship to `remainderAvailable` ?

Reply:
`remainderAvailable` is used below. So that was the reason it is not collapses into the single modification using `%~`